components/org.wso2.carbon.identity.oauth.uma.permission.service/src/main/java/org/wso2/carbon/identity/oauth/uma/permission/service/impl/PermissionServiceImpl.java

```diff
@@ -65,10 +65,13 @@ public PermissionTicketModel issuePermissionTicket(List<Resource> resourceList,
     public List<Resource> validateAccessToken(String accessToken) throws UMAClientException, UMAServerException {
 
         try {
-            AccessTokenDO tokenDO = OAuth2Util.getAccessTokenDOfromTokenIdentifier(accessToken);
-            String tokenId = tokenDO.getTokenId();
-            String permissionTicket = PermissionTicketDAO.retrievePermissionTicketForTokenId(tokenId);
-            return PermissionTicketDAO.getResourcesForPermissionTicket(permissionTicket);
+            AccessTokenDO tokenDO = OAuth2Util.findAccessToken(accessToken, false);
+            if (tokenDO != null) {
+                String tokenId = tokenDO.getTokenId();
+                String permissionTicket = PermissionTicketDAO.retrievePermissionTicketForTokenId(tokenId);
+                return PermissionTicketDAO.getResourcesForPermissionTicket(permissionTicket);
+            }
+            return null;
```

Comment:
Wouldnt it be good to return an empty array

Reply:
Fixed with https://github.com/wso2-extensions/identity-oauth-uma/pull/78/commits/22641d7ef77acae7935c7f1f9deddac9ed5399a5